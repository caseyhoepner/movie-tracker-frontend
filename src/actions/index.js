export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
})

export const toggleFavorite = (movieId) => ({
  type: 'TOGGLE_FAVORITE',
  movieId: movieId
})

export const setUser = (userId) => ({
  type: 'SET_USER',
  userId: userId
})

export const removeUser = () => ({
  type: 'REMOVE_USER'
})

export const filterMovies = (movies) => ({
  type: 'FILTER_MOVIES',
  movies
})