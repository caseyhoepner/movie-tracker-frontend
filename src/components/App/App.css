.App {
  text-align: center;
}

.header-container {
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
  margin: 0 7%;
}

.logo {
  display: flex;
  align-items: flex-end;
}

.logo-icon {
  height: 150px;
}