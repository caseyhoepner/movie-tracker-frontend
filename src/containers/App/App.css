.a-App {
  font-family: "Futura", sans-serif;
  letter-spacing: 0.5px;
  box-sizing: border-box;
  color: #D4A931;
}
.a-header-container {
  display: flex;
  align-items: flex-end;
  justify-content: space-around;
  padding-bottom: 20px;
  height: 120px;
  background-color: #48788C;
  box-shadow: 1px 1px 10px 4px #4c6c7c;
}
.a-logo {
  width: 400px;
  display: flex;
  justify-content: space-around;
  align-items: flex-end;
  font-weight: 700;
}
.a-logo-icon {
  height: 100px;
}

.a-header-title {
  font-size: 3em;
}