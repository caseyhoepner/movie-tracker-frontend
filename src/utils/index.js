export const fetchMovies = async (apiKey) => {
  const wesAPI = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1&with_people=5655`;
  const response = await fetch(wesAPI);
  if (!response.ok) {
      throw new Error(response.statusText);
  } else {
    const movieData = await response.json();
    const cleanData = cleanMovies(movieData.results)
    return cleanData
  }
}

const cleanMovies = (results) => {
  return results.map(movie => {
    return {
      movie_id: movie.id,
      overview: movie.overview,
      poster_path: movie.poster_path,
      release_date: movie.release_date,
      title: movie.title,
      vote_average: movie.vote_average,
      isFavorite: false
    }
  })  
}

export const addFavorite = async (movie) => {
  const url = 'http://localhost:3000/api/users/favorites/new'
  const response = await fetch(url, {
    method: 'POST',
    body: JSON.stringify(movie),
    headers: { 'Content-Type': 'application/json' }
  });
  const result = await response.json();
  console.log('Success:', JSON.stringify(result))
}

export const addUser = async (user) => {
  const dataBaseAPI = 'http://localhost:3000/api/users/new';

  try { 
    const response = await fetch(dataBaseAPI, {
      method: 'POST',
      body: JSON.stringify(user),
      headers: {
        'Content-Type': 'application/json'
      }
    })
    if (!response.ok) {
      throw new Error(response.statusText);
    } 
    return await response.json();
  } catch (error){
    return 'Email has already been used';
  }
}

export const loginUser = async (user) => {
  const dataBaseAPI = 'http://localhost:3000/api/users';
  const email = user.email.toLowerCase();
  try {
    const response = await fetch(dataBaseAPI, {
      method: 'POST',
      body: JSON.stringify({email, password: user.password}),
      headers: {
        'Content-Type': 'application/json'
      }
    })
    if (!response.ok) {
      throw new Error(response.statusText);
    } else {
      return await response.json();
    } 
  } catch (error) {
    return 'Email and password do not match.'
  }
}

export const fetchFavorites = async (userId) => {
  const dataBaseAPI = `http://localhost:3000/api/users/${userId}/favorites`;
  const response = await fetch(dataBaseAPI);
  const result = await response.json()
  return result.data
}

