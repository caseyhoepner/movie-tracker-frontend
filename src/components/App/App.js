import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        WesTracker
      </div>
    );
  }
}

export default App;
