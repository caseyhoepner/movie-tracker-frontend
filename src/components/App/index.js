import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { addMovies } from '../../actions';
import Nav from '../Nav';
import wesIcon from '../../assets/images/wes.png';
import './App.css';
import * as API from '../../utils';
import apiKey from '../../API-key.js';
import MovieContainer from '../MovieContainer';

class App extends Component {
  async componentDidMount() {
    const movies = await API.fetchMovies(apiKey)
    await this.props.addMovies(movies)
  }

  render() {
    return (
      <div className="App">
        <header className="header-container">
          <div className="logo">
            <img
              src={ wesIcon }
              className="logo-icon"
              alt="An icon of Wes Anderson's face"
            />
            <h1>WesTracker</h1>
          </div>
          <Nav />
        </header>
        <Route
          exact path='/'
          component={ MovieContainer }
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies))
})

export default connect(null, mapDispatchToProps)(App);
