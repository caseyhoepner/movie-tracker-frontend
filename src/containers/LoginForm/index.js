import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux';
import { addUser } from '../../actions';
import * as API from '../../utils'
import './LoginForm.css'

export class LoginForm extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: '',
      isSigningUp: false
    }
  }

  handleInputChange = (event) => {
    const { name, value } = event.target
    this.setState({[name]: value})
  }

  handleSubmit = (event) => {
    event.preventDefault()
  }

  toggleSigningUp = () => {
    this.setState({isSigningUp: !this.state.isSigningUp})
  }

  render() {
    const { isSigningUp, email, password } = this.state

    return(
      <form onSubmit={this.handleSubmit}>
        <h2>
          { isSigningUp ? 'ENLIST IN TEAM ZISSOU' : 'HAVE AN ACCOUNT?' }
        </h2>
        <div className="input-container">
          <input
            className="email-login"
            type="email"
            name="email"
            value={email}
            onChange={this.handleInputChange}
            placeholder= {
              isSigningUp ? 'klaus@daimler.net' : "wes@anderson.com"
            }
          />
          <input
            className="password-login"
            type="password"
            name="password"
            value={password}
            onChange={this.handleInputChange}
            placeholder="Type password here"
          />
        </div>
        <button
          className="submit-login"
          onClick={this.handleSubmit}
        >
          { isSigningUp ? 'CREATE ACCOUNT' : 'LOGIN' }
        </button>
        <div className="sign-up-container">
          <h3>
            { isSigningUp ? 'ERM, NVM...' : 'NEED AN ACCOUNT?' }
          </h3>
          <button
            className="sign-up-btn"
            onClick={this.toggleSigningUp}
          >
            { isSigningUp ? 'CANCEL SIGN UP' : 'SIGN UP' }
          </button>
        </div>
      </form>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addUser: (email, password, id) => dispatch(addUser(email, password, id))
})

LoginForm.propTypes = {
  addUser: PropTypes.func.isRequired
}

export default connect(null, mapDispatchToProps)(LoginForm);