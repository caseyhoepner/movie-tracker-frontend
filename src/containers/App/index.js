import React, { Component } from 'react';
import { NavLink, Route } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as API from '../../utils';
import apiKey from '../../API-key.js';
import { addMovies } from '../../actions';
import MovieContainer from '../MovieContainer';
import Nav from '../../components/Nav';
import LoginForm from '../LoginForm'
import wesIcon from '../../assets/images/wes.png';
import './App.css';

export class App extends Component {
  async componentDidMount() {
    const movies = await API.fetchMovies(apiKey)
    await this.props.addMovies(movies)
  }

  render() {
    const { user } = this.props
    return (
      <div className="App">
        <header className="header-container">
          <NavLink to='/' className="logo">
            <img
              src={ wesIcon }
              className="logo-icon"
              alt="An icon of Wes Anderson's face"
            />
            <h1>WesTracker</h1>
          </NavLink>
          <Nav userLoggedIn={user.email ? true : false} />
        </header>
        <Route
          exact path='/'
          component={ MovieContainer }
        />
        <Route
          exact path='/login'
          component={ LoginForm }
        />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies))
})

App.propTypes = {
  user: PropTypes.object.isRequired,
  addMovies: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
