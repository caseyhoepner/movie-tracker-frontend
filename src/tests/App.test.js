import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps, mapStateToProps } from '../containers/App';
import { mockUser, mockMovies, mockFavMovieId } from './testMocks';
import * as API from '../utils';
import * as Actions from '../actions'

describe('App', () => {
  let wrapper;
  let mockAddMovies;

  beforeEach(() => {
    mockAddMovies = jest.fn();
    wrapper = shallow(<App addMovies={mockAddMovies} user={{}} />);
  })

  describe('App Component', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the addMovies action', () => {
      const mockDispatch = jest.fn()
      const expected = Actions.addMovies(mockMovies)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addMovies(mockMovies)

      expect(mockDispatch).toHaveBeenCalledWith(expected)
    })
  })

  describe('mapStateToProps', () => {
    it('should parse the moves from state', () => {
      const mockState = {
        user: mockUser,
        movies: mockMovies
      }

      const expected = mockMovies

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps.movies).toEqual(expected)
    })
  })
})