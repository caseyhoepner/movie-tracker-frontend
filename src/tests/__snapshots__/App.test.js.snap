// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="header-container"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="logo"
      to="/"
    >
      <img
        alt="An icon of Wes Anderson's face"
        className="logo-icon"
        src="wes.png"
      />
      <h1>
        WesTracker
      </h1>
    </NavLink>
    <Nav
      userLoggedIn={false}
    />
  </header>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/login"
  />
</div>
`;
